pkg/build/frontend.go

```diff
@@ -378,9 +378,36 @@ func solvePlatform(ctx context.Context, bopts *BOpts, pl ocispecs.Platform, c ga
 		return nil, nil, err
 	}
 
-	// This only happens when the dockerfile is just `FROM scratch`
+	// Handle metadata-only builds (ENV/ARG/LABEL without RUN/COPY/ADD)
 	if ref == nil {
-		return nil, nil, ErrNoBuildDirectives
+		if img == nil {
+			return nil, nil, ErrNoBuildDirectives
+		}
+
+		// OCI manifests require a layers array (cannot be null)
+		stateWithLayer := llb.Scratch().
+			File(llb.Mkfile("/.container-metadata-only", 0644, []byte("# This image contains only metadata (ENV/ARG/LABEL)\n"))).
```

Comment:
Use a constant for marker path, that way it is self-documenting

```swift
const metadataOnlyMarker = "/.container-metadata-only"
```

use that const instead of the literal